Source/MediaSampleProvider.h

```diff
@@ -155,42 +159,42 @@ namespace FFmpegInteropX
 
         LONGLONG ConvertDuration(TimeSpan duration)
         {
-            return LONGLONG(duration.Duration / timeBaseFactor);
+            return LONGLONG(duration.count() / timeBaseFactor);
         }
 
-    protected private:
+    protected:
         MediaSampleProvider(
-            FFmpegReader^ reader,
+            std::shared_ptr<FFmpegReader> reader,
             AVFormatContext* avFormatCtx,
             AVCodecContext* avCodecCtx,
-            MediaSourceConfig^ config,
+            MediaSourceConfig const& config,
             int streamIndex,
             HardwareDecoderStatus hardwareDecoderStatus);
 
     private:
         std::queue<AVPacket*> m_packetQueue;
-        int64 m_nextPacketPts;
-        IMediaStreamDescriptor^ m_streamDescriptor;
+        INT64 m_nextPacketPts = 0;
+        winrt::Windows::Media::Core::IMediaStreamDescriptor m_streamDescriptor = nullptr;
         HardwareDecoderStatus hardwareDecoderStatus;
 
-    internal:
+    public:
```

Comment:
All members that were once "internal" are now public. This is because C++/winRT does not support internal members. I don't see how we could achieve the same effect.

Reply:
My point is that it was wrong to have them internal, already in the old implementation. We should not directly access fields from outside the class. Fields should be private or protected, but not internal or public.

Reply:
For fields which are both read and write, we shouldn't encapsulate, it would be just noise.
I think we inherited many internal stuff from the original repo.